script/generate_missing_latest_version_views

```diff
@@ -0,0 +1,131 @@
+    default=[DEFAULT_EXCLUDE],
+    metavar="project:dataset.table",
+    help="Latest-version views that should be ignored, may use shell-style wildcards,"
+    f" defaults to: {DEFAULT_EXCLUDE}",
+)
+parser.add_argument(
+    "--log-level",
+    default="INFO",
+    help="Defaults to INFO",
+)
+
+args = parser.parse_args()
+
+# set log level
+try:
+    logging.basicConfig(level=args.log_level, format="%(levelname)s %(message)s")
+except ValueError as e:
+    parser.error(f"argument --log-level: {e}")
+
+wildcards = set("*?[]")
+def uses_wildcards(pattern: str) -> bool:
+    return bool(set("*?[]") & set(pattern))
+
+patterns = [
+    (project, dataset, table)
+    for pattern in args.patterns
+    for project, _, dataset_table in (pattern.partition(":"),)
+    for dataset, _, table in (dataset_table.partition("."),)
+]
+all_projects = None
+all_datasets = {}
+all_tables = {}
+views = {}
+for pattern in args.patterns:
+    project, _, dataset_table = pattern.partition(":")
+    dataset, _, table = dataset_table.partition(".")
+    projects = [project or client.project]
+    dataset = dataset or "*"
+    table = table or "*"
+    if uses_wildcards(project):
+        if all_projects is None:
+            all_projects = {p.project_id for p in client.list_projects()}
+        projects = [p for p in all_projects if fnmatchcase(project, p)]
+    for project in projects:
+        datasets = [dataset]
+        if uses_wildcards(dataset):
+            if project not in all_datasets:
+                all_datasets[project] = {
+                    d.dataset_id for d in client.list_datasets(project)
+                }
+            datasets = [d for d in all_datasets[project] if fnmatchcase(d, dataset)]
+        for dataset in datasets:
+            dataset = f"{project}.{dataset}"
+            tables = [(f"{dataset}.{table}", None)]
+            if uses_wildcards(table):
+                if dataset not in all_tables:
+                    all_tables[dataset] = list(client.list_tables(dataset))
+                tables = [
+                    (f"{dataset}.{t.table_id}", t.table_type)
+                    for t in all_tables[dataset] if fnmatchcase(t.table_id, table)
+                ]
+            for full_table_id, table_type in tables:
+                view = VERSION_RE.sub("", full_table_id)
+                if view not in views:
+                    views[view] = {}
+                views[view][full_table_id] = table_type
+
+for view, tables in views.items():
+    if any(fnmatchcase(pattern, view) for pattern in args.exclude):
+        log.info("skipping table: matched by exclude pattern: {view}")
+        continue
+    version = max(
+        int(match.group()[2:])
+        for table in tables
+        for match in (VERSION_RE.search(table),)
+        if match is not None
+    )
+    project, dataset, viewname = view.split(".")
+    target = f"{view}_v{version}"
+    view_query = f"SELECT * FROM\n  `{target}`"
+    view_dataset = dataset[:-7]
```

Comment:
presumably the `-7` here is to clip off `_stable`, but I think this should handle other input datasets, e.g. `_derived`

not sure if it's better to match exact suffixes:
```suggestion
    view_dataset = re.sub("_(derived|live|raw|stable)$", "", dataset)
```
or simply assume there is some suffix:
```suggestion
    view_dataset = dataset.rsplit("_", 1)[-1]
```